pocket_coffea/utils/plot_utils.py

```diff
@@ -200,65 +301,149 @@ def get_systematic_uncertainty(
 
         # Sum in quadrature of the systematic uncertainties for each MC sample
-        syst_err2_tot_up += syst_err2_up
-        syst_err2_tot_down += syst_err2_down
+        #syst_err2_tot_up += syst_err2_up
+        #syst_err2_tot_down += syst_err2_down
 
-    return np.sqrt(syst_err2_tot_up), np.sqrt(syst_err2_tot_down)
+    #return np.sqrt(syst_err2_tot_up), np.sqrt(syst_err2_tot_down)
+    return syst_err2_up_dict,  syst_err2_down_dict
 
 
-def plot_systematic_uncertainty(
-    stack_mc_nominal, syst_err_up, syst_err_down, ax, ratio=False
-):
-    '''This function plots the asymmetric systematic uncertainty band on top of the MC stack, if `ratio` is set to False.
-    To plot the systematic uncertainty in a ratio plot, `ratio` has to be set to True and the uncertainty band will be plotted around 1 in the ratio plot.'''
-    h_mc_sum = stack_sum(stack_mc_nominal)
+def plot_uncertainty_band(h_mc_sum, syst_err2_up, syst_err2_down, ax, ratio=False):
     nom = h_mc_sum.values()
-    up = nom + syst_err_up
-    down = nom - syst_err_down
+    # Sum in quadrature of the systematic uncertainties for each variation
+    up = nom + np.sqrt(sum(syst_err2_up.values()))
+    down = nom - np.sqrt(sum(syst_err2_down.values()))
 
     if ratio:
         up = up / nom
         down = down / nom
 
-    # unc_band = np.array([down, up])
-    unc_band = np.nan_to_num(np.array([down, up]), nan=1)
-
-    # print("up", up)
-    # print("nom", nom)
-    # print("down", down)
+    unc_band = np.array([down, up])
 
     ax.fill_between(
         h_mc_sum.axes[0].edges,
         np.r_[unc_band[0], unc_band[0, -1]],
         np.r_[unc_band[1], unc_band[1, -1]],
-        **opts_unc,
+        **opts_unc['total'],
         label="syst. unc.",
     )
+    if ratio:
+        ax.hlines(1.0, *ak.Array(h_mc_sum.axes[0].edges)[[0,-1]], colors='gray', linestyles='dashed')
+
+def plot_systematic_uncertainty(
+    stack_mc_nominal, syst_err2_up, syst_err2_down, ax, ratio=False, split_systematics=False, only_syst=[], partial_unc_band=False,
+):
+    '''This function plots the asymmetric systematic uncertainty band on top of the MC stack, if `ratio` is set to False.
+    To plot the systematic uncertainty in a ratio plot, `ratio` has to be set to True and the uncertainty band will be plotted around 1 in the ratio plot.'''
+    if (not type(syst_err2_up) == dict) or (not type(syst_err2_down) == dict):
+        raise Exception("`syst_err2_up` and `syst_err2_down` must be dictionaries")
+
+    h_mc_sum = stack_sum(stack_mc_nominal)
+    nom = h_mc_sum.values()
+
+    if split_systematics and only_syst:
+        syst_err2_up_filtered = { syst : unc for syst, unc in syst_err2_up.items() if any(key in syst for key in only_syst)}
+        syst_err2_down_filtered = { syst : unc for syst, unc in syst_err2_down.items() if any(key in syst for key in only_syst)}
+
+    if not split_systematics:
+        plot_uncertainty_band(h_mc_sum, syst_err2_up, syst_err2_down, ax, ratio)
```

Comment:
this is not using the options `only_syst`. 
I think that the function must be coherent with the argument. If `only_syst` is given, also the plot with the total uncertainty should only contain the filtered systematic.

Reply:
This is fixed with a patch commit: https://github.com/PocketCoffea/PocketCoffea/pull/60/commits/2e3e318a91165114535b818139a84feea627eadd